crates/uv/tests/pip_compile.rs

```diff
@@ -6708,6 +6708,138 @@ fn universal_multi_version() -> Result<()> {
+            .arg("requirements.in")
+            .arg("--universal"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    # This file was autogenerated by uv via the following command:
+    #    uv pip compile --cache-dir [CACHE_DIR] requirements.in --universal
+    cmake==3.28.4 ; platform_machine == 'x86_64' and platform_system == 'Linux'
+        # via triton
+    filelock==3.13.1
+        # via
+        #   torch
+        #   triton
+    jinja2==3.1.3
+        # via torch
+    lit==18.1.2 ; platform_machine == 'x86_64' and platform_system == 'Linux'
+        # via triton
+    markupsafe==2.1.5
+        # via jinja2
+    mpmath==1.3.0
+        # via sympy
+    networkx==3.2.1
+        # via torch
+    sympy==1.12
+        # via torch
+    torch==2.0.0+cpu ; platform_machine != 'x86_64'
+        # via -r requirements.in
+    torch==2.0.0+cu118 ; platform_machine == 'x86_64'
+        # via
+        #   -r requirements.in
+        #   triton
+    triton==2.0.0 ; platform_machine == 'x86_64' and platform_system == 'Linux'
+        # via torch
+    typing-extensions==4.10.0
+        # via torch
+
+    ----- stderr -----
+    Resolved 12 packages in [TIME]
+    "###
+    );
+
+    Ok(())
+}
+
+#[test]
+fn universal_transitive_disjoint_locals() -> Result<()> {
+    let context = TestContext::new("3.12");
+    let requirements_in = context.temp_dir.child("requirements.in");
+    requirements_in.write_str(indoc::indoc! {r"
+        --find-links https://download.pytorch.org/whl/torch_stable.html
+
+        torch==2.0.0+cu118 ; platform_machine == 'x86_64'
+        torch==2.0.0+cpu ; platform_machine != 'x86_64'
+        torchvision==0.15.1
+    "})?;
+
+    uv_snapshot!(context.filters(), windows_filters=false, context.pip_compile()
+            .arg("requirements.in")
+            .arg("--universal"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    # This file was autogenerated by uv via the following command:
+    #    uv pip compile --cache-dir [CACHE_DIR] requirements.in --universal
+    certifi==2024.2.2
+        # via requests
+    charset-normalizer==3.3.2
+        # via requests
+    cmake==3.28.4 ; platform_machine == 'x86_64' and platform_system == 'Linux'
+        # via triton
+    filelock==3.13.1
+        # via
+        #   torch
+        #   triton
+    idna==3.6
+        # via requests
+    jinja2==3.1.3
+        # via torch
+    lit==18.1.2 ; platform_machine == 'x86_64' and platform_system == 'Linux'
+        # via triton
+    markupsafe==2.1.5
+        # via jinja2
+    mpmath==1.3.0
+        # via sympy
+    networkx==3.2.1
+        # via torch
+    numpy==1.26.4
+        # via torchvision
+    pillow==10.2.0
+        # via torchvision
+    requests==2.31.0
+        # via torchvision
+    sympy==1.12
+        # via torch
+    torch==2.0.0+cpu
+        # via
+        #   -r requirements.in
+        #   torchvision
+    torch==2.0.0+cu118
```

Comment:
I think this is wrong right, because of #5086? If you could leave a comment on the test noting that, that would be great.